.github/workflows/ci.yml

```diff
@@ -88,6 +93,10 @@ jobs:
         if: ${{ inputs.run_dependency_convergence }}
         run: echo "MVN_DEPENDENCY_CONVERGENCE=-Dflink.convergence.phase=install -Pcheck-convergence" >> $GITHUB_ENV
 
+      - name: "Disable archunit tests"
+        if: ${{ inputs.skip_archunit_tests }}
+        run: echo "MVN_ARCHUNIT_TESTS=-Dtest='!*ArchitectureTest'" >> $GITHUB_ENV
```

Comment:
I tried similar settings locally like
`mvn -Dflink.version=1.18.0  -Dtest=\!*ArchitectureTest* clean install` and for dirs without archunit tests it fails like
```
No tests were executed!  (Set -DfailIfNoTests=false to ignore this error.) 
```
then if we add mentioned flag to the command then surefire mixes integration and non integration tests (which does not happen without this flag)

would be great if this is somehow covered here...

Reply:
@snuyanzin thanks for your feedback, I'll take a look.

Reply:
the idea which seems working at least for my setup with opensearch connector is using `<include>`, `<exclude>` for surefire plugin and different profiles like
```xml
<properties>
    	<exclude.tests>nothing-to-exclude</exclude.tests>
</properties>
...
<plugin>
		<groupId>org.apache.maven.plugins</groupId>
		<artifactId>maven-surefire-plugin</artifactId>
		<configuration>
			...
		</configuration>
		<executions>
			<execution>
                            <configuration>
	                            <includes>
		                            <include>${test.unit.pattern}</include>
	                            </includes>
	                            <excludes>
		                            <exclude>${exclude.tests}</exclude>
	                            </excludes>
                                   ...
                            </configuration>
                      ...		
	              </execution>
              </executions>
</plugin>
...
	<profiles>
		<profile>
			<id>noArchUnit</id>
			<properties>
				<exclude.tests>**/org/apache/flink/architecture/**/*.java</exclude.tests>
			</properties>
		</profile>
	</profiles>
...
```

Reply:
First of all `-Dtest` is not a surefire parameter but a maven one. 

Then, the above command is not meant to be run from anywhere but from the CI at the root of a connector module that has the flink standard connector structure (the one of test project sample) including the architecture tests.

To be sure, I ran  `mvn install -Dflink.version=1.18.0 -Dtest='!*ArchitectureTest'` from [the cassandra connector root](https://github.com/apache/flink-connector-cassandra/tree/main/flink-connector-cassandra):

1. with all tests in the structure:
=> UTests and ITests are run and archunit tests are skipped as expected (see log [1] below)

2. after removing the archunit tests as you mentioned something about dirs without these tests:
=> UTests and ITests are run as expected (see log [2] below)

==> I cannot reproduce what you observed


[1]
```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.flink:flink-connector-cassandra_2.12:jar:4.0-SNAPSHOT
[WARNING] 'artifactId' contains an expression but should be a constant. @ org.apache.flink:flink-connector-cassandra_${scala.binary.version}:4.0-SNAPSHOT, /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/pom.xml, line 32, column 14
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ----------< org.apache.flink:flink-connector-cassandra_2.12 >-----------
[INFO] Building Flink : Connectors : Cassandra 4.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.2:check (validate) @ flink-connector-cassandra_2.12 ---
[WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
[WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
[INFO] You have 0 Checkstyle violations.
[INFO] 
[INFO] --- spotless-maven-plugin:2.27.1:check (spotless-check) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (forbid-direct-table-planner-dependencies) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven-version) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.5:process (process-resource-bundles) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ flink-connector-cassandra_2.12 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 0 resource
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ flink-connector-cassandra_2.12 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ flink-connector-cassandra_2.12 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ flink-connector-cassandra_2.12 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] Running org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.283 s - in org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 s - in org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 s - in org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 s - in org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.436 s - in org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.jboss.netty.util.internal.ByteBufferUtil (file:/tmp/flink-rpc-akkaf91156d2-7ae8-4d71-bd4f-dced7d4a75e1.jar) to method java.nio.DirectByteBuffer.cleaner()
[ERROR] WARNING: Please consider reporting this to the maintainers of org.jboss.netty.util.internal.ByteBufferUtil
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 56.932 s - in org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 105.701 s - in org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 43, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ flink-connector-cassandra_2.12 ---
[INFO] Building jar: /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.2.4:shade (shade-flink) @ flink-connector-cassandra_2.12 ---
[INFO] Including com.datastax.cassandra:cassandra-driver-core:jar:shaded:3.11.2 in the shaded jar.
[INFO] Including com.datastax.cassandra:cassandra-driver-mapping:jar:3.11.2 in the shaded jar.
[INFO] Including com.google.guava:guava:jar:19.0 in the shaded jar.
[INFO] Excluding io.dropwizard.metrics:metrics-core:jar:3.2.2 from the shaded jar.
[INFO] No artifact matching filter org.apache.flink:flink-shaded-force-shading
[INFO] No artifact matching filter io.netty:netty
[WARNING] cassandra-driver-core-3.11.2-shaded.jar, cassandra-driver-mapping-3.11.2.jar, flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar, guava-19.0.jar define 1 overlapping resource: 
[WARNING]   - META-INF/MANIFEST.MF
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://maven.apache.org/plugins/maven-shade-plugin/
[INFO] Replacing original artifact with shaded artifact.
[INFO] Replacing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar with /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT-shaded.jar
[INFO] Dependency-reduced POM written at: /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/dependency-reduced-pom.xml
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (integration-tests) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.075 s - in org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.975 s - in org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.jboss.netty.util.internal.ByteBufferUtil (file:/tmp/flink-rpc-akkaf6320afa-293e-4879-8871-c4fc3058a2cd.jar) to method java.nio.DirectByteBuffer.cleaner()
[ERROR] WARNING: Please consider reporting this to the maintainers of org.jboss.netty.util.internal.ByteBufferUtil
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 56.696 s - in org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Running org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.07 s - in org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.081 s - in org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.085 s - in org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 105.637 s - in org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 43, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- japicmp-maven-plugin:0.16.0_m325:cmp (default) @ flink-connector-cassandra_2.12 ---
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.diff'.
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.xml'.
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.html'.
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ flink-connector-cassandra_2.12 ---
[INFO] Installing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar to /home/echauchot/.m2/repository/org/apache/flink/flink-connector-cassandra_2.12/4.0-SNAPSHOT/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar
[INFO] Installing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/dependency-reduced-pom.xml to /home/echauchot/.m2/repository/org/apache/flink/flink-connector-cassandra_2.12/4.0-SNAPSHOT/flink-connector-cassandra_2.12-4.0-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:45 min
[INFO] Finished at: 2023-10-27T18:27:09+02:00
[INFO] ------------------------------------------------------------------------

```

[2]

```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.flink:flink-connector-cassandra_2.12:jar:4.0-SNAPSHOT
[WARNING] 'artifactId' contains an expression but should be a constant. @ org.apache.flink:flink-connector-cassandra_${scala.binary.version}:4.0-SNAPSHOT, /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/pom.xml, line 32, column 14
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ----------< org.apache.flink:flink-connector-cassandra_2.12 >-----------
[INFO] Building Flink : Connectors : Cassandra 4.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.2:check (validate) @ flink-connector-cassandra_2.12 ---
[WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
[WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
[INFO] You have 0 Checkstyle violations.
[INFO] 
[INFO] --- spotless-maven-plugin:2.27.1:check (spotless-check) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (forbid-direct-table-planner-dependencies) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven-version) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.5:process (process-resource-bundles) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ flink-connector-cassandra_2.12 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 0 resource
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ flink-connector-cassandra_2.12 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ flink-connector-cassandra_2.12 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ flink-connector-cassandra_2.12 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] Running org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.446 s - in org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 s - in org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.014 s - in org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.018 s - in org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.659 s - in org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.jboss.netty.util.internal.ByteBufferUtil (file:/tmp/flink-rpc-akkabdb2b72e-22fa-4100-ac6b-aa9126fcdcee.jar) to method java.nio.DirectByteBuffer.cleaner()
[ERROR] WARNING: Please consider reporting this to the maintainers of org.jboss.netty.util.internal.ByteBufferUtil
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 57.532 s - in org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 106.445 s - in org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 43, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ flink-connector-cassandra_2.12 ---
[INFO] Building jar: /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.2.4:shade (shade-flink) @ flink-connector-cassandra_2.12 ---
[INFO] Including com.datastax.cassandra:cassandra-driver-core:jar:shaded:3.11.2 in the shaded jar.
[INFO] Including com.datastax.cassandra:cassandra-driver-mapping:jar:3.11.2 in the shaded jar.
[INFO] Including com.google.guava:guava:jar:19.0 in the shaded jar.
[INFO] Excluding io.dropwizard.metrics:metrics-core:jar:3.2.2 from the shaded jar.
[INFO] No artifact matching filter org.apache.flink:flink-shaded-force-shading
[INFO] No artifact matching filter io.netty:netty
[WARNING] cassandra-driver-core-3.11.2-shaded.jar, cassandra-driver-mapping-3.11.2.jar, flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar, guava-19.0.jar define 1 overlapping resource: 
[WARNING]   - META-INF/MANIFEST.MF
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://maven.apache.org/plugins/maven-shade-plugin/
[INFO] Replacing original artifact with shaded artifact.
[INFO] Replacing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar with /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT-shaded.jar
[INFO] Dependency-reduced POM written at: /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/dependency-reduced-pom.xml
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (integration-tests) @ flink-connector-cassandra_2.12 ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.145 s - in org.apache.flink.streaming.connectors.cassandra.CassandraSinkBaseTest
[INFO] Running org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.072 s - in org.apache.flink.streaming.connectors.cassandra.CassandraTupleWriteAheadSinkTest
[INFO] Running org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.jboss.netty.util.internal.ByteBufferUtil (file:/tmp/flink-rpc-akka20cb269c-4523-4f18-9763-af2c15df2c6a.jar) to method java.nio.DirectByteBuffer.cleaner()
[ERROR] WARNING: Please consider reporting this to the maintainers of org.jboss.netty.util.internal.ByteBufferUtil
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 56.508 s - in org.apache.flink.connector.cassandra.source.CassandraSourceITCase
[INFO] Running org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.068 s - in org.apache.flink.connector.cassandra.source.split.CassandraSplitSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 s - in org.apache.flink.connector.cassandra.source.enumerator.CassandraEnumeratorStateSerializerTest
[INFO] Running org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.086 s - in org.apache.flink.connector.cassandra.source.reader.CassandraQueryTest
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 105.905 s - in org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 43, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- japicmp-maven-plugin:0.16.0_m325:cmp (default) @ flink-connector-cassandra_2.12 ---
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.diff'.
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.xml'.
[INFO] Written file '/home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/${rootDir}/tools/japicmp-output/flink-connector-cassandra_2.12/japicmp/japicmp.html'.
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ flink-connector-cassandra_2.12 ---
[INFO] Installing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar to /home/echauchot/.m2/repository/org/apache/flink/flink-connector-cassandra_2.12/4.0-SNAPSHOT/flink-connector-cassandra_2.12-4.0-SNAPSHOT.jar
[INFO] Installing /home/echauchot/projects/flink-connector-cassandra/flink-connector-cassandra/target/dependency-reduced-pom.xml to /home/echauchot/.m2/repository/org/apache/flink/flink-connector-cassandra_2.12/4.0-SNAPSHOT/flink-connector-cassandra_2.12-4.0-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:46 min
[INFO] Finished at: 2023-10-27T18:40:44+02:00
[INFO] ------------------------------------------------------------------------

```

Reply:
> the idea which seems working at least for my setup with opensearch connector is using `<include>`, `<exclude>` for surefire plugin and different profiles like
> 
> ```
> <properties>
>     	<exclude.tests>nothing-to-exclude</exclude.tests>
> </properties>
> ...
> <plugin>
> 		<groupId>org.apache.maven.plugins</groupId>
> 		<artifactId>maven-surefire-plugin</artifactId>
> 		<configuration>
> 			...
> 		</configuration>
> 		<executions>
> 			<execution>
>                            <!--execute all the unit tests-->
> 			   <id>default-test</id>			
>                             <configuration>
>                             
> 	                            <includes>
> 		                            <include>${test.unit.pattern}</include>
> 	                            </includes>
> 	                            <excludes>
> 		                            <exclude>${exclude.tests}</exclude>
> 	                            </excludes>
>                                    ...
>                             </configuration>
>                       ...		
> 	              </execution>
> 	              <execution>
> 		      <!--execute all the integration tests-->
> 		      <id>integration-tests</id>
> 		      ...
> 		      <configuration>
> 			      <includes>
> 				      <include>**/*.*</include>
> 			      </includes>
> 			      <excludes>
> 				      <exclude>${test.unit.pattern}</exclude>
> 				      <exclude>${exclude.tests}</exclude>
> 			      </excludes>
> 			      ...
> 		      </configuration>
> 	      </execution>
>               </executions>
> </plugin>
> ...
> 	<profiles>
> 		<profile>
> 			<id>noArchUnit</id>
> 			<properties>
> 				<exclude.tests>**/org/apache/flink/architecture/**/*.java</exclude.tests>
> 			</properties>
> 		</profile>
> 	</profiles>
> ...
> ```

This solution is the first I proposed but it requires having a profile for exclusions in the pom of all the connectors so it is a breaking change and the solution was abandoned toward the -Dtest that does not require a change to the poms.
See my comment above -Dtest works as expected.

Reply:
> See my comment above -Dtest works as expected.

It doesn't. To be more explicit than @snuyanzin, this will fail in flink-sql-connector-* modules. Naturally it works in the Cassandra repo because it doesn't have a sql-connector module.

Surefire exclusions are the way to go, but we can't see these on the command-line since it'd override the unit/it test separation setup in the parent pom.
I'd suggest to add another property to the connector parent pom `additionalExcludes`, which it adds do the surefire exclusions.

Reply:
> It doesn't. To be more explicit than @snuyanzin, this will fail in flink-sql-connector-* modules since they contain no tests. Naturally it works in the Cassandra repo because it doesn't have a sql-connector module, nor any module without any tests.

Ah, the failure was about a module with no tests at all, it was written "without archunit tests". Indeed, in that case it does not work. Thanks for the precision Chesnay.

> Surefire exclusions are the way to go, but we can't set these on the command-line since it'd override the unit/it test separation setup in the parent pom.
I'd suggest to add another property to the connector parent pom (e.g., additionalExcludes), which it adds to the surefire exclusions.
This should then "just work".

Yes, at first I submitted [this PR](https://github.com/apache/flink-connector-shared-utils/pull/22/files) that added surefire exclusions to the parent pom by adding a custom profile but it had this drawback indeed. Of course I can add a property to add to already defined exclusions. The drawback I see that I wanted to avoid is that it modifies the parent pom leading to the need to release flink-connector-parent and to bump its version in the connectors. But I guess this trade-off is acceptable considering the sql modules problem above.

Reply:
I have updated the PR as discussed above:
- use new **additionalExcludes** property to exclude archunit tests. This property is added in [this PR](https://github.com/apache/flink-connector-shared-utils/pull/26) that needs to be merged and released (v 1.0.1) before merging the current PR. See [here](https://github.com/apache/flink-connector-shared-utils/pull/26#issue-1992394911) how it was tested.
- move to **org.apache.flink**:flink-connector-parent instead of **io.github.zentol.flink**:flink-connector-parent in test project.
- changed the test project structure to include 2 submodules. Test-project is now a parent and there is 1 module with tests (the former test project)  and one without tests. This is to illustrate the case discussed above of the sql module containing no tests.

@zentol PTAL